src/Workspaces/Core/Portable/Rename/ConflictEngine/ConflictResolver.cs

```diff
@@ -175,12 +175,39 @@ private static bool IsRenameValid(ConflictResolution conflictResolution, ISymbol
             if (renamedSymbol.ContainingSymbol is INamedTypeSymbol)
             {
                 var otherThingsNamedTheSame = renamedSymbol.ContainingType.GetMembers(renamedSymbol.Name)
-                                                       .Where(s => !s.Equals(renamedSymbol) && string.Equals(s.MetadataName, renamedSymbol.MetadataName, StringComparison.Ordinal) &&
-                                                              (s.Kind != SymbolKind.Method || renamedSymbol.Kind != SymbolKind.Method)).ToList();
+                                                       .Where(s => !s.Equals(renamedSymbol) && 
+                                                                   string.Equals(s.MetadataName, renamedSymbol.MetadataName, StringComparison.Ordinal) &&
+                                                                   (s.Kind != SymbolKind.Method || renamedSymbol.Kind != SymbolKind.Method));
 
                 AddConflictingSymbolLocations(otherThingsNamedTheSame, conflictResolution, reverseMappedLocations);
             }
 
+
+            if (renamedSymbol is INamespaceSymbol && renamedSymbol.ContainingSymbol is INamespaceSymbol)
```

Comment:
Yeah, I was in Rome and doing as the Romans were doing, but I don't like it either. Will adjust.